src/main.rs

```diff
@@ -16,8 +17,57 @@ use localauth0::{controller, APP_NAME};
 // Singleton logger. Used to free user from manually passing Logger objects around.
 static LOGGER_GUARD: GuardLoggerCell = GuardLoggerCell::new();
 
+fn main() -> Result<(), Box<dyn Error>> {
+    match std::env::args().nth(1).as_deref() {
+        Some("healthcheck") => Ok(healthcheck()?),
+        _ => Ok(server()?),
+    }
+}
+
+async fn is_endpoint_healthy(client: &reqwest::Client, endpoint: String) -> bool {
+    let status = client.get(&endpoint).send().await.map(|r| r.error_for_status());
+
+    match status {
+        Ok(Ok(_)) => {
+            println!("{endpoint} OK");
+            true
+        }
+        _ => {
+            println!("{endpoint} ERROR {status:?}");
+            false
+        }
+    }
+}
+
+fn healthcheck() -> Result<(), String> {
+    tokio::runtime::Builder::new_current_thread()
```

Comment:
This is another reason why i prefer having a different binary

Reply:
Wdym?